lib/msf/core/exploit/http/client.rb

```diff
@@ -88,6 +89,9 @@ def initialize(info = {})
     )
     register_autofilter_ports([ 80, 8080, 443, 8000, 8888, 8880, 8008, 3000, 8443 ])
     register_autofilter_services(%W{ http https })
+
+    # Initialize an empty cookie jar to keep cookies
+    self.cookie_jar = Set.new
```

Comment:
For future travellers, this implementation was superseded by
https://github.com/rapid7/metasploit-framework/pull/14831